src/flow.py

```
@@ -590,10 +595,15 @@ def process_notifications(self, timeout_secs=0.05, sid=0):
         LOG.debug("process_notifications start")
         while self._loop_process_notifications.is_set():
             try:
-                self.sessions[sid].consume_notification(timeout_secs)
-            except Exception as exception:
-                LOG.debug("consume_notification failed: %s", exception)
+                session = self.sessions[sid]
+            except KeyError as key_error:
+                LOG.debug("no session %s", key_error)
                 break
+            try:
+                session.consume_notification(timeout_secs)
+            except:
+                # Log error and keep looping
+                LOG.exception("consume_notification failed")
```

This doesn't really print the traceback, just silently fails.